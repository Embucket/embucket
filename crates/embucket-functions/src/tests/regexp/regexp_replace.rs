use crate::test_query;

test_query!(
    regexp_replace_basic,
    "SELECT REGEXP_REPLACE('nevermore1, nevermore2, nevermore3.', 'nevermore', 'moreover')",
    snapshot_path = "regexp_replace"
);

test_query!(
    regexp_replace_remove,
    "SELECT REGEXP_REPLACE('It was the best of times, it was the worst of times',
                      '( ){1,}',
                      '')",
    snapshot_path = "regexp_replace"
);

test_query!(
    regexp_replace_occurrence,
    "SELECT REGEXP_REPLACE('It was the best of times, it was the worst of times',
                      'times',
                      'days',
                      1,
                      2)",
    snapshot_path = "regexp_replace"
);
