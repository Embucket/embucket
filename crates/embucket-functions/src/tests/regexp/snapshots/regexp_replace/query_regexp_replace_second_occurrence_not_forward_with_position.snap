---
source: crates/embucket-functions/src/tests/regexp/regexp_replace.rs
description: "\"SELECT REGEXP_REPLACE('It was the best of times, it was the worst of times. times',\n                      'times',\n                      'days',\n                      30,\n                      2) AS result;\""
---
Ok(
    [
        "+-----------------------------------------------------------+",
        "| result                                                    |",
        "+-----------------------------------------------------------+",
        "| It was the best of times, it was the worst of times. days |",
        "+-----------------------------------------------------------+",
    ],
)
